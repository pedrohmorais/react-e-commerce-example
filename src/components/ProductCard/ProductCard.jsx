import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Card,
  Button,
} from '@catho/quantum';

import { addToChart } from '../ShopCart/actions';

import {
  PriceBlock,
  ImgContainer,
  StyledFooter,
  CTAButton,
  CTAButtonContainer,
} from './styles';

// eslint-disable-next-line react/prefer-stateless-function
class ProductCard extends React.Component {
  render() {
    const {
      id,
      name,
      slug,
      imgUrl,
      formatedPrice,
      addToChartAction,
    } = this.props;

    const handleToChartClick = () => {
      addToChartAction({
        id,
        name,
        slug,
        imgUrl,
        formatedPrice,
      });
      // eslint-disable-next-line no-alert
      alert('Produto adicionado com sucesso!');
    };

    return (
      <Card>
        <Card.Header>
          <Card.HeaderText>
            <Card.Title small>{name}</Card.Title>
          </Card.HeaderText>
        </Card.Header>
        <PriceBlock>
          {formatedPrice}
        </PriceBlock>
        <ImgContainer src={imgUrl} alt={slug} />
        <StyledFooter>
          <CTAButtonContainer>
            <Button as="a" href={`/produto/${slug}`} size="small" skin="secondary">
              Ver detalhes
            </Button>
          </CTAButtonContainer>
          <div>
            <CTAButton size="small" skin="primary" icon="shopping_cart" onClick={() => handleToChartClick()}>
              Adicionar ao carrinho
            </CTAButton>
          </div>
        </StyledFooter>
      </Card>
    );
  };
}

ProductCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
  formatedPrice: PropTypes.string.isRequired,
  addToChartAction: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  addToChartAction: filters => dispatch(addToChart(filters)),
});
export default connect(
  null,
  mapDispatchToProps,
)(ProductCard);
