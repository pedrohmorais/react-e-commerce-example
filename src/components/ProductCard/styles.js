import styled from 'styled-components';
import {
  Card,
  Colors,
} from '@catho/quantum';

const PriceBlock = styled(Card.Content)`
  font-weight: bold;
  font-size: 18px;
`;
const ImgContainer = styled(Card.Media)`
  margin: 12px 15px 20px;
  background-color: ${Colors.neutral[0]};
`;

export {
  PriceBlock,
  ImgContainer,
}