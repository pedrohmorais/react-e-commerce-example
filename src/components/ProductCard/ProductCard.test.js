import React from 'react';
import renderer from 'react-test-renderer';

import ProductsService from 'services/Products';
import { ProductCard } from './ProductCard';

describe('<ProductCard />', () => {
  it('should match snapshot', () => {
    const product = ProductsService.getById(1);
    product.addToCartAction = jest.fn();

    expect(
      renderer.create(<ProductCard {...product} />).toJSON(),
    ).toMatchSnapshot();
  });
});
