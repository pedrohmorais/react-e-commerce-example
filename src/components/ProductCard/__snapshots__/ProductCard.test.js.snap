// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
.c14 {
  pointer-events: none;
  margin-right: 8px;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  font-size: 12px;
  line-height: 18px;
  min-height: 32px;
  padding: 4px 16px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #DE0059;
  color: #ffffff;
  border: 2px solid #DE0059;
  box-shadow: 0px 3px 1px -2px rgba(153,153,153,0.2),0px 2px 2px 0px rgba(153,153,153,0.14),0px 1px 5px 0px rgba(153,153,153,0.12);
}

.c10:hover {
  box-shadow: 0px 2px 4px -1px rgba(158,0,63,0.2),0px 4px 5px 0px rgba(158,0,63,0.14),0px 1px 10px 0px rgba(158,0,63,0.12);
  background-color: #9E003F;
  border-color: #9E003F;
}

.c10:focus,
.c10:focus-within {
  box-shadow: 0px 2px 4px -1px rgba(222,0,89,0.2),0px 4px 5px 0px rgba(222,0,89,0.14),0px 1px 10px 0px rgba(222,0,89,0.12);
}

.c10:active {
  box-shadow: 0px 5px 5px -3px rgba(158,0,63,0.2),0px 8px 10px 1px rgba(158,0,63,0.14),0px 3px 14px 2px rgba(158,0,63,0.12);
  background-color: #9E003F;
}

.c10 .c13 function (_ref5) {
  -webkit-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -moz-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -ms-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  return: "\\n          font-size: ".concat(fontSizes[size],";\\n          margin-right: ").concat(xxsmall,"px;\\n        ");
}

.c10 .c13 function (_ref5) var fontSizes = {
  xsmall: "".concat(medium,"px"), small:"".concat(medium,"px"), medium:"".concat(large,"px"), large:"".concat(large,"px"), xlarge:"".concat(large,"px");
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  font-size: 12px;
  line-height: 18px;
  min-height: 32px;
  padding: 4px 16px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #1250C4;
  color: #ffffff;
  border: 2px solid #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(153,153,153,0.2),0px 2px 2px 0px rgba(153,153,153,0.14),0px 1px 5px 0px rgba(153,153,153,0.12);
}

.c11:hover {
  box-shadow: 0px 2px 4px -1px rgba(0,47,123,0.2),0px 4px 5px 0px rgba(0,47,123,0.14),0px 1px 10px 0px rgba(0,47,123,0.12);
  background-color: #002F7B;
  border-color: #002F7B;
}

.c11:focus,
.c11:focus-within {
  box-shadow: 0px 2px 4px -1px rgba(18,80,196,0.2),0px 4px 5px 0px rgba(18,80,196,0.14),0px 1px 10px 0px rgba(18,80,196,0.12);
}

.c11:active {
  box-shadow: 0px 5px 5px -3px rgba(0,47,123,0.2),0px 8px 10px 1px rgba(0,47,123,0.14),0px 3px 14px 2px rgba(0,47,123,0.12);
  background-color: #002F7B;
}

.c11 .c13 function (_ref5) {
  -webkit-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -moz-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -ms-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  return: "\\n          font-size: ".concat(fontSizes[size],";\\n          margin-right: ").concat(xxsmall,"px;\\n        ");
}

.c11 .c13 function (_ref5) var fontSizes = {
  xsmall: "".concat(medium,"px"), small:"".concat(medium,"px"), medium:"".concat(large,"px"), large:"".concat(large,"px"), xlarge:"".concat(large,"px");
}

.c5 {
  padding-bottom: 56.25%;
  position: relative;
  overflow: hidden;
  margin-top: 12px;
  background-color: #e0e0e0;
}

.c7 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 16px 0;
}

.c1 > * {
  margin-right: 16px;
}

.c1 > *:last-child {
  margin-right: 0;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

.c3 {
  font-size: 20px;
  font-weight: 600;
  margin: 0;
}

.c0 {
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  box-shadow: 0px 2px 1px -1px rgba(66,66,66,0.2),0px 1px 1px 0px rgba(66,66,66,0.14),0px 1px 3px 0px rgba(66,66,66,0.12);
  background-color: #ffffff;
}

.c4 {
  margin: 0;
  padding: 12px 16px;
  font-size: 14px;
  font-weight: bold;
  font-size: 18px;
}

.c6 {
  margin: 12px 15px 20px;
  background-color: #ffffff;
}

.c8 {
  padding: 0 16px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  min-height: 36px;
  margin-right: 20px;
}

.c9 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
}

.c9 button {
  min-height: 36px;
  margin-right: 20px;
}

<article
  className="c0"
>
  <header
    className="c1"
  >
    <div
      className="c2"
    >
      <h2
        className="c3"
      >
        M92 WE Silver Edition
      </h2>
    </div>
  </header>
  <div
    className="c4"
  >
    R$Â 999.99
  </div>
  <div
    className="c5 c6"
  >
    <img
      alt="m92-we-silver-edition"
      className="c7"
      src="/static/products/r-1/m92-we-silver-edition.jpg"
    />
  </div>
  <footer
    className="c8"
  >
    <div
      className="c9"
    >
      <a
        className="c10"
        disabled={false}
        href="/produto/m92-we-silver-edition"
        onClick={[Function]}
        size="small"
        type="button"
      >
        
        Ver detalhes
      </a>
    </div>
    <div>
      <button
        className="c11 c12"
        disabled={false}
        onClick={[Function]}
        size="small"
        type="button"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root c13 c14"
          size="small"
          style={
            Object {
              "color": "",
            }
          }
          theme={
            Object {
              "baseFontSize": 16,
              "breakpoints": Object {
                "large": Object {
                  "columns": 12,
                  "width": 1440,
                },
                "medium": Object {
                  "columns": 12,
                  "width": 1024,
                },
                "small": Object {
                  "columns": 8,
                  "width": 600,
                },
                "xlarge": Object {
                  "columns": 12,
                  "width": 1920,
                },
                "xsmall": Object {
                  "columns": 4,
                },
              },
              "colors": Object {
                "error": Object {
                  "100": "#fff5f5",
                  "300": "#f88e86",
                  "500": "#ec584e",
                  "700": "#e91b0c",
                  "900": "#ac001a",
                },
                "neutral": Object {
                  "0": "#ffffff",
                  "100": "#f2f2f2",
                  "1000": "#000000",
                  "300": "#e0e0e0",
                  "500": "#999999",
                  "700": "#424242",
                  "900": "#191919",
                },
                "primary": Object {
                  "100": "#E5EDFC",
                  "300": "#89AAE7",
                  "500": "#0CC0EA",
                  "700": "#1250C4",
                  "900": "#002F7B",
                },
                "secondary": Object {
                  "100": "#F7B0C8",
                  "300": "#F278A1",
                  "500": "#E91E63",
                  "700": "#DE0059",
                  "900": "#9E003F",
                },
                "success": Object {
                  "100": "#dcedc8",
                  "300": "#c9eda2",
                  "500": "#7ed321",
                  "700": "#3b610f",
                  "900": "#004D40",
                },
                "warning": Object {
                  "100": "#ffefd6",
                  "300": "#ffda6a",
                  "500": "#ffc107",
                  "700": "#f09100",
                  "900": "#d14900",
                },
              },
              "components": Object {
                "button": Object {
                  "skins": Object {
                    "error": Object {
                      "mainColor": Object {
                        "100": "#fff5f5",
                        "300": "#f88e86",
                        "500": "#ec584e",
                        "700": "#e91b0c",
                        "900": "#ac001a",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                    "neutral": Object {
                      "mainColor": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                    "primary": Object {
                      "mainColor": Object {
                        "100": "#E5EDFC",
                        "300": "#89AAE7",
                        "500": "#0CC0EA",
                        "700": "#1250C4",
                        "900": "#002F7B",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                    "secondary": Object {
                      "mainColor": Object {
                        "100": "#F7B0C8",
                        "300": "#F278A1",
                        "500": "#E91E63",
                        "700": "#DE0059",
                        "900": "#9E003F",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                    "success": Object {
                      "mainColor": Object {
                        "100": "#dcedc8",
                        "300": "#c9eda2",
                        "500": "#7ed321",
                        "700": "#3b610f",
                        "900": "#004D40",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                    "warning": Object {
                      "mainColor": Object {
                        "100": "#ffefd6",
                        "300": "#ffda6a",
                        "500": "#ffc107",
                        "700": "#f09100",
                        "900": "#d14900",
                      },
                      "text": Object {
                        "0": "#ffffff",
                        "100": "#f2f2f2",
                        "1000": "#000000",
                        "300": "#e0e0e0",
                        "500": "#999999",
                        "700": "#424242",
                        "900": "#191919",
                      },
                    },
                  },
                },
              },
              "spacing": Object {
                "large": 24,
                "medium": 16,
                "small": 12,
                "xlarge": 32,
                "xsmall": 8,
                "xxlarge": 40,
                "xxsmall": 4,
                "xxxlarge": 48,
                "xxxsmall": 2,
              },
            }
          }
        >
          shopping_cart
        </span>
        Adicionar ao carrinho
      </button>
    </div>
  </footer>
</article>
`;
